### ISSUE TITLE
Nested Group Support

### ISSUE THREAD

**Issue by timhirsh** [github/saml-ldap-team-sync#21]
Our org has some nested AD groups.  Here's an example group structure:
```
Group A
  * User 1
  * Group B

Group B
  * User 2
  * User 3
```

If I were to sync `Group A` to a GitHub team, I would like to see Users 1, 2, and 3 to be added to the team.  Currently I'm seeing an exception when encountering the `Group B` member because it doesn't have a `sAMAccountName`.

**Comment by primetheus** [comment]
The exceptions are handled in 2.0 and it skips over groups. We'll be looking into support for nested teams in version 2.1

**Comment by sudhir-korde-tm** [comment]
Hi @primetheus , is there any update on this enhancement.

**Comment by primetheus** [comment]
@sudhir-korde-tm we can only implement a portion of this, as GitHub only allows for one parent team. For example:

The following mapping is possible with GitHub

```
GroupA:
    Group1:
        User1
        User2
    Group2:
        User1
        User5
GroupB:
    Group3:
        User3
        User4
```

In this scenario each team has _one_ parent team.

---

The following mapping is _not_ possible with GitHub

```
GroupA:
    Group1:
        User1
        User2
    Group2:
        User1
        User5
GroupB:
    Group1:
        User1
        User2
```

In this scenario we are attempting to add `Group1` to both `GroupA` and `GroupB`. This is not something that GitHub supports with team nesting. 

## Options

What we _can_ do, is flatten out the nested groups in the IdP so they return individual users, and add them individually to the parent team. This is less than perfect, but it would be doable. @sudhir-korde-tm @tim-tk is this something that would be a sufficient stop-gap?

**Comment by tim-tk** [comment]
@primetheus Yes, flattening the groups would be a valid solution for us! Thank you!
